typing-practice-react/src/index.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import {displayStats, displayTodayStats, displayTotalStats, toggleExtendedStats} from './displayStats.js';
+import {parseLocalStorage, resetLocalStorageForTodayStats, resetLocalStorageForTotalStats} from './localStorageHandler.js';
+import {incrementSeconds} from './timeHandler.js';
+import {handleKeyDownEvent} from './handleKeyDownEvent.js';
+import {generateFinalText} from './generateFinalText.js';
+import {markCurrentChar} from './displayText.js'
+import {resetProgress} from './resetProgress.js'
+
+let localStorage = window.localStorage;
+localStorage = parseLocalStorage(localStorage);
+const keyboardKeysArray = [['A', 0, 0], ['B', 0, 0], ['C', 0, 0], ['D', 0, 0], ['E', 0, 0], ['F', 0, 0], ['G', 0, 0], ['H', 0, 0], ['I', 0, 0], ['J', 0, 0], ['K', 0, 0], ['L', 0, 0], ['M', 0, 0],
+                           ['N', 0, 0], ['O', 0, 0], ['P', 0, 0], ['Q', 0, 0], ['R', 0, 0], ['S', 0, 0], ['T', 0, 0], ['U', 0, 0], ['V', 0, 0], ['W', 0, 0], ['X', 0, 0], ['Y', 0, 0], ['Z', 0, 0]];
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.statsTextForSeconds = React.createRef();
+    this.todayStatsText = React.createRef();
+    this.totalStatsText = React.createRef();
+    this.lastSetStatsText = React.createRef();
+    this.state = {
+      userMistakesCount: 0,
+      userKeyTypeCount: 0,
+      seconds: 0,
+      charIndex: 0,
+      tryCounter: 0,
+      amountOfSets: 0,
+      incrementSecondsInterval: null,
+      generateTextButtonIsClicked: false,
+      extendedStatsString: "",
+    };
+  }
+
+  beginCounting(state, statsTextForSeconds, todayStatsText, totalStatsText, lastSetStatsText) {
+    this.incrementSecondsInterval = setInterval(function() {
+      state.seconds = incrementSeconds(state.seconds, statsTextForSeconds, todayStatsText, totalStatsText);
```

Comment:
Use `this.setState` for changing the state :)

Reply:
🤯